sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/reflection/KalixMethod.scala

```diff
@@ -181,6 +180,13 @@ case class SyntheticRequestServiceMethod(
       "Missing request method mapping. Kalix Java SDK methods must have a request method defined.")
   }
 
+  if (javaMethod.getParameterAnnotations.toSet.flatten
+      .collect { case (p: PathVariable) => p }
+      .exists(_.required() == false)) {
+
+    throw ServiceIntrospectionException(javaMethod, "Currently all @PathVariables must be defined as required.")
+  }
```

Comment:
annoying that we need `collect` and `exists`. 

Maybe something like: 

```scala
javaMethod.getParameterAnnotations.toSet.flatten.exists {
  case p: PathVariable => !p.required
  case _ => false
}
```

Not sure if make it more readable though. 
Maybe if we move it out of the `if`?

Reply:
Personally, I think that collect and exists is more readable, so I will leave it as it is.